conformance/utils/http/http.go

```diff
@@ -286,8 +286,38 @@ func CompareRequest(cReq *roundtripper.CapturedRequest, cRes *roundtripper.Captu
 		if expected.RedirectRequest == nil {
 			return nil
 		}
-		if expected.RedirectRequest.Hostname != cRes.RedirectRequest.Hostname {
-			return fmt.Errorf("expected redirected hostname to be %s, got %s", expected.RedirectRequest.Hostname, cRes.RedirectRequest.Hostname)
+
+		// if the expected host is nil it means we do not test host redirect.
+		// in that case we are setting it to the one we got from the response because we do not know the ip/host of the gateway.
+		if expected.RedirectRequest.Host == "" {
+			expected.RedirectRequest.Host = cRes.RedirectRequest.Host
+		}
+
+		if expected.RedirectRequest.Port == "" {
+			expected.RedirectRequest.Port = req.URL.Port()
+		}
+
+		if expected.RedirectRequest.Scheme == "" {
+			expected.RedirectRequest.Scheme = req.URL.Scheme
+		}
+
+		if expected.RedirectRequest.Path == "" {
+			expected.RedirectRequest.Path = req.URL.Path
+		}
+		if expected.RedirectRequest.Host != cRes.RedirectRequest.Host {
```

Comment:
Tiny nit: All the conditions above this are setting reasonable defaults, all the conditions below are checking values. I'd expect there to be more separation, maybe even comments, but in this case, this transition is the only one without an extra newline.

Reply:
added a separate function